src/lib/format/l20n/parser.js

```diff
@@ -5,201 +5,126 @@ import { L10nError } from '../../errors';
 const MAX_PLACEABLES = 100;
 
 export default {
-  _patterns: null,
-
-  init: function() {
-    this._patterns = {
-      index: /@cldr\.plural\(\$?(\w+)\)/g,
-      placeables: /\{\{\s*\$?([^\s]*?)\s*\}\}/,
-      unesc: /\\({{|u[0-9a-fA-F]{4}|.)/g,
-    };
-  },
-
-  parse: function (emit, string, simple) {
-    if (!this._patterns) {
-      this.init();
-    }
+  parse: function(env, string) {
     this._source = string;
     this._index = 0;
-    this._length = this._source.length;
-    this.simpleMode = simple;
-    this.emit = emit;
+    this._length = string.length;
+    this.entries = Object.create(null);
 
-    return this.getL20n();
+    return this.getResource();
```

Comment:
I wonder if this could be a good opportunity to refactor the parser a bit by not using `this` at all. Right now, invoking the `parse` method requires you to bind it to the `*Parser` object.  What if we stored `source`, `index`, `length` and `entries` in a new object , `state`, created when `parse` is called, and passed it into every function: `getResource(state)`?